src/app/test/db/mongo/models_test.py

```diff
@@ -242,6 +242,50 @@ def test_template_check(self):
             RequestTemplateSchema.get_attribute_names()
         )
 
+    @pytest.fixture()
+    def request_model(self):
+        req = Request(
+            system="foo",
+            command="bar",
+            status="CREATED",
+            system_version="1.0.0",
+            instance_name="foobar",
+            namespace="barfoo",
+        )
+        req.parameters = {"message": "hi"}
+        req.output = "bye"
+        req.parameters_gridfs.put = Mock()
+        req.output_gridfs.put = Mock()
+        return req
+
+    def test_save_stores_in_gridfs_after_maxsize(self, request_model):
+        request_model.parameters = {"message": "a" * 6 * 1_000_000}
```

Comment:
The tests you have here are generally fine, but I suggest we tweak them so that they can work independent of the actual `max_param_size` and `max_output_size` values.  There are two main reasons to do this:

* The tests as written are a bit too dependent on what the actual max size settings happen to be, which isn't really what you care about of the purposes of the test. This means that if someone came along and changed those max sizes to 7MB instead of 5MB, the tests now fail.  The 5MB limit isn't what we're trying to test, it's that whatever the limit is, it gets enforced.

* Generating a 6MB string is all fast and fine, but what if the limit was 500MB? Or 5GB? Generating a sufficiently large object in those cases wouldn't be great.

So I think what you want to do instead is move those max size variables somewhere mockable (constants at the top of the file maybe?) and then mock the values in your tests to be some arbitrary, small value that your tests are in control of.

Reply:
I moved these to a constant at the beginning of `models.py`. I kept the encoding there, though, as I'm pretty sure the encoding will never need to be changed from UTF-8 for our purposes.